base/socket.jl

```diff
@@ -2,6 +2,7 @@
 abstract IPAddr
 
 Base.isless{T<:IPAddr}(a::T, b::T) = isless(a.host, b.host)
+Base.convert{T<:Integer}(dt::Type{T}, ip::IPAddr) = dt(ip.host)
```

Comment:
The typeassert `dt::Type{T}` is satisfied by an object `dt` such that `isa(dt, Type{T})`. For all `T`, there is exactly one object--the singleton type object--which is of type `Type{T}` which is `T` itself. Therefore, `dt === T`.